apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,803 @@
+            _isRepresentativeOf(_voter, _representative) &&
+            _hasNotVotedYet(vote_, _voter) &&
+            !_withinOverruleWindow(vote_);
+    }
+
+    /**
+    * @dev Return the main information for vote #`_voteId`
+    *      Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Vote identifier
+    * @return Vote executed status
+    * @return Vote start date
+    * @return Vote snapshot block
+    * @return Vote support required
+    * @return Vote minimum acceptance quorum
+    * @return Vote overrule window period
+    * @return Vote yeas amount
+    * @return Vote nays amount
+    * @return Vote power
+    * @return Vote script
+    */
+    function getVote(uint256 _voteId)
+        external
+        view
+        voteExists(_voteId)
+        returns (
+            bool executed,
+            uint64 startDate,
+            uint64 snapshotBlock,
+            uint64 supportRequired,
+            uint64 minAcceptQuorum,
+            uint64 voteOverruleWindow,
+            uint256 yea,
+            uint256 nay,
+            uint256 votingPower,
+            bytes script
+        )
+    {
+        Vote storage vote_ = votes[_voteId];
+
+        executed = _isExecuted(vote_);
+        startDate = vote_.startDate;
+        snapshotBlock = vote_.snapshotBlock;
+        supportRequired = vote_.supportRequiredPct;
+        minAcceptQuorum = vote_.minAcceptQuorumPct;
+        voteOverruleWindow = vote_.overruleWindow;
+        yea = vote_.yea;
+        nay = vote_.nay;
+        votingPower = vote_.votingPower;
+        script = vote_.executionScript;
+    }
+
+    /**
+    * @dev Return the disputable information for vote #`_voteId`
+    *      Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Vote identifier
+    * @return Vote agreement action ID
+    * @return Vote paused date
+    * @return Vote paused duration
+    * @return Vote status
+    */
+    function getVoteDisputableInfo(uint256 _voteId)
+        external
+        view
+        voteExists(_voteId)
+        returns (uint256 actionId, uint64 pausedAt, uint64 pauseDuration, VoteStatus status)
+    {
+        Vote storage vote_ = votes[_voteId];
+        actionId = vote_.actionId;
+        pausedAt = vote_.pausedAt;
+        pauseDuration = vote_.pauseDuration;
+        status = vote_.status;
+    }
+
+    /**
+    * @dev Tells whether vote #`_voteId` is open
+    *      Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Id for vote
+    * @return True if the vote is still open
+    */
+    function isVoteOpen(uint256 _voteId) external view voteExists(_voteId) returns (bool) {
```

Comment:
The base voting app was a lot simpler, but with the new paused state here, I wonder if we should be more explicit with this function's name (renaming it to `isVoteOpenForVoting()` or etc.) to differentiate it from an open vote that is paused.

Reply:
I like that!